packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -448,6 +489,7 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
       const rejectedTag = rejected ? 'reject' : 'fulfill';
       // prettier-ignore
       msgs.push(`resolve:${rejectedTag}:${rpid}${mapSlots()};${data.body}`);
+      enqueueRemotePromiseIDRetirement(remoteID, rpid);
```

Comment:
I think we should remove the outbound clist entry at this point (maybe cleanest to do it inside `enqueueRemotePromiseIDRetirement`, which then might want to be named `beginRemotePromiseIDRetirement` or something).

By sending a `resolve`, we're promising to never mention this rpid again, even though we tolerate *receipt* of the rpid for a while (until this `resolve` is acked), because we know our peer might have message on the wire.

If some other outbound message references this recently-resolved promise, we must make a new (ephemeral) rpid for it. Our `provideRemoteForLocal` will only do that if the lpid is not already in the outbound clist. If we don't remove the outbound clist entry now, that message will re-use the old rpid, and we've already promised to not mention it again.

The *inbound* clist entry must remain in place until the retirement handshake completes (with the receipt of the ack).

Reply:
I think that's right.  I'll fix.  One hygiene question: does it make sense, at the we get the ack and remove the inbound entry, to put in an assert that verifies that there was no outbound entry?

Reply:
I think yes, although we need to remember to ignore the ack if we'd re-exported the item in the meantime, which means the outbound entry will be back.